dali/operators/math/expressions/expression_impl_cpu.h

```diff
@@ -53,46 +57,105 @@ class ExprImplCpuT : public ExprImplBase {
 template <ArithmeticOp op, typename Result, typename Left, typename Right>
 class ExprImplCpuTT : public ExprImplBase {
  public:
-  void Execute(ExprImplContext &ctx, const std::vector<ExtendedTileDesc> &tiles,
-               TileRange range) override {
-    assert(range.begin + 1 == range.end &&
+  void Execute(ExprImplContext &ctx,
+               span<const SampleDesc> samples,
+               span<const TileDesc> tiles) override {
+    assert(tiles.size() == 1 &&
            "CPU Expression implementation can handle only one tile at a time");
-    const auto &tile = tiles[range.begin];
-    auto output = static_cast<Result *>(tile.output);
-    auto left = static_cast<const Left *>(tile.args[0]);
-    auto right = static_cast<const Right *>(tile.args[1]);
-    Execute(output, left, right, tile.desc.extent_size);
+    const auto &tile = tiles[0];
+    const auto &sample = samples[tile.sample_idx];
+    auto &output = sample.output;
+    auto *output_ptr = static_cast<Result *>(output.data);
+    auto &left = sample.args[0];
+    const auto *left_ptr = static_cast<const Left *>(sample.args[0].data);
+    auto &right = sample.args[1];
+    const auto *right_ptr = static_cast<const Right *>(sample.args[1].data);
+
+    if (sample.args[0].shape == sample.args[1].shape) {
```

Comment:
I wonder, can we have a comment somewhere describing when the implementation picks broadcasting variant and when just processes linear tiles. I know that one can look at it as the implementation detail, but it is quite far from producing the linear tiles vs full tiles intended for broadcasting, so I think some comment pointing from ther in this direction (or the inverse) would probably be apperciated :)